bucket/foobar2000.json

```diff
@@ -28,22 +29,28 @@
             "    New-Item \"$persist_dir\\playlists\" -ItemType Directory -Force | Out-Null",
             "    if (Test-Path \"$env:Appdata\\foobar2000\") {",
             "        Write-Host -F yellow \"Copying old '$env:Appdata\\foobar2000' to '$persist_dir'\"",
-            "        Copy-Item \"$env:Appdata\\foobar2000\\*\" \"$dir\" -Recurse -Force",
+            "        Copy-Item \"$env:Appdata\\foobar2000\\*\" \"$dir\" -Force -Recurse",
+            "        if (Test-Path \"$env:Appdata\\foobar2000\\playlists-v*\") {",
+            "            Copy-Item \"$dir\\playlists-v*\" \"$persist_dir\\playlists\" -Force -Recurse",
+            "        }",
+            "        if (Test-Path \"$env:Appdata\\foobar2000\\user-components\") {",
+            "            Copy-Item \"$dir\\user-components\" \"$persist_dir\\user-components\" -Force -Recurse",
+            "        }",
             "    }",
             "}",
             "else {",
-            "    $shortVersion = $version.Split('.')[0, 1] -Join '.'",
-            "    Copy-Item \"$persist_dir\\playlists\" \"$dir\\playlist-v$shortVersion\" -Force -Recurse",
+            "    Copy-Item \"$persist_dir\\playlists\\*\" \"$dir\" -Force -Recurse",
             "}",
             "if (Test-Path \"$persist_dir\\theme.fth\") {",
             "    Copy-Item \"$persist_dir\\theme.fth\" \"$dir\" -Force",
             "}"
         ]
     },
+    "pre_install": "if(!(Test-Path \"$persist_dir\\user-components\")) { New-Item \"$dir\\user-components\" -ItemType Directory -Force | Out-Null}",
     "post_install": "if (is_directory \"$dir\\theme.fth\") { Remove-Item \"$dir\\theme.fth\" -Force }",
     "uninstaller": {
         "script": [
-            "Copy-Item \"$dir\\playlists-v*\\*\" \"$persist_dir\\playlists\" -Force  -Recurse",
+            "Copy-Item \"$dir\\playlists-v*\" \"$persist_dir\\playlists\" -Force  -Recurse",
```

Comment:
This will result in

`$persist_dir\playlists\playlists-vXXX.XX`. Is it correct?

Reply:
Yes. Foobar2000 uses `$dir\playlists-vX.X` when it reads playlists. There is no way to support arbitrary version numbers with just scoop's the `persist`.

Reply:
So why do you make it nested? Will foobar2000 rename playlist-v1.3 to playlist-v1.4 when update?

Reply:
It doesn't actually need to be nested in the persist directory - I was just keeping the folder from the manifest as I had found it before.

Reply:
Well, I still don't understand... And why you copy it twice in `installer` and `uninstaller` ?

Reply:
I'm not sure I understand what you mean copied twice?

The first `if` section in the installer only runs if it's a fresh installation via coop, and possibly runs a migration from local AppData if found there. The `playlists` directory isn't actually a part of the foobar2000 installation, but is used to check if it's a fresh installation via scoop. The else section and uninstaller ensures the `playlists-vX.X` folders are carried over between updates (if `-p` isn't used).

Reply:
I mean, why we need to copy the playlist to persist_dir in installer? We will copy it in uninstaller when update.

Reply:
Oh sorry, I thought you were talking about L39. I see what you mean, I'll remove the redundant installer copy.